iOS/DuckDuckGo/DuckPlayer/NativeUI/DuckPlayerNativeUIPresenter.swift

```diff
@@ -418,15 +475,28 @@ extension DuckPlayerNativeUIPresenter: DuckPlayerNativeUIPresenting {
         // Show the container view if it's not already visible
         if !containerViewModel.sheetVisible {
             containerViewModel.show()
+            postPillVisibilityNotification(isVisible: true)
         }
     }
 
     /// Dismisses the currently presented entry pill
     @MainActor
-    func dismissPill(reset: Bool = false, animated: Bool = true) {
+    func dismissPill(reset: Bool = false, animated: Bool = true, programatic: Bool = true) {
         // First reset constraints immediately
         resetWebViewConstraint()
         
+        postPillVisibilityNotification(isVisible: false)
+
+        // If was dismissed by the user, increment the dismiss count
+        if !programatic {
+            appSettings.duckPlayerPillDismissCount += 1
+
+            if appSettings.duckPlayerPillDismissCount >= 3 {
```

Comment:
Using a constant for the threshold, with the Asana task where this number was decided included as a comment, is probably good for documentation. I can see this changing in the future, and it would be helpful to easily get the context when needed